app/models/workers_scaling/workers_scaling_algorithm_runner.rb

```diff
@@ -0,0 +1,48 @@
+##
+# Scheduler to run WorkersScalingAlgorithm decision loop periodically
+class WorkersScalingAlgorithmRunner
+
+  ##
+  # Expects:
+  # * instance of Experiment
+  # * instance of WorkersScalingAlgorithm for given Experiment
+  # * interval between decision loop executions in seconds
+  def initialize(experiment, algorithm, interval)
+    @experiment = experiment
+    @algorithm = algorithm
+    @interval = interval
+  end
+
+  ##
+  # Initializes WorkersScalingAlgorithm, then starts its decision loop
+  # Stops when there is no next execution time set
+  def start
+    Thread.new do
+      @algorithm.initial_deployment
+      @next_execution_time = Time.now + interval
+
+      until @next_execution_time.nil?
+
+        while @next_execution_time > Time.now do
+          sleep @next_execution_time - Time.now
+        end
+
+        execute_and_schedule
+      end
+    end
+  end
+
+  ##
+  # Executes decision loop of WorkersScalingAlgorithm, then schedules next execution
+  # Next execution time is not set when Experiment is completed
+  def execute_and_schedule
```

Comment:
Chcemy żeby ta metoda działała tylko w jednym egzemplarzu naraz. Mutex w rubym pozwala sprawdzić, czy nie jest w danym momencie zajęty przez inny wątek. Zrobić tak, że (1) czekamy na wolny mutex i natychmiast wchodzimy, czy (2) mutex jest zajęty, więc nie będziemy nic robić?

(1) - jeżeli experiment_status_check będzie trwać więcej, niż interval, to może wystąpić taka sytuacja, jak na obrazku (jedno zaschedulowanie będzie zignorowane) - wydaje mi się że to jest w porządku: 
![ex int](https://cloud.githubusercontent.com/assets/3855041/9653023/e6f40002-5220-11e5-90b5-b2cd60142950.jpg)
To samo się wydarzy właściwie zawsze, kiedy synchroniczne wywołanie będzie musiało czekać na zakończenie asynchronicznego.
W normalnym przypadku będzie tak jak chcemy:
![int ex](https://cloud.githubusercontent.com/assets/3855041/9653822/71673d12-5226-11e5-8b59-3bdca4e370c6.jpg)


(2) - wydaje mi się, że wtedy w ogóle nie zareagujemy na wydarzenie, bo aktualny przebieg pętli nie będzie miał wiadomości o tym wydarzeniu.